src/stabilization_guide.md

```diff
@@ -0,0 +1,181 @@
+features, declared below in a separate call to `declare_features!`.
+When it is done, it should look like:
+
+    ```
+    // pub(restricted) visibilities (RFC 1422)
+    (accepted, pub_restricted, "1.31.0", Some(32409)),
+    // ^^^^^^ note that we changed this
+    ```
+
+Note that, the version number is updated to be the version number
+of the stable release where this feature will appear. This can be
+found by consulting https://forge.rust-lang.org/, which will guide
+you the next stable release number. You want to add 1 to that,
+because the code that lands today will become go into beta on that
+date, and then become stable after that. So, at the time of this
+writing, the next stable release (what is currently beta, iow) was
+1.30.0, hence I wrote 1.31.0 above.
+
+### Removing existing uses of the feature-gate
+
+Next search for the feature string (in this case, pub_restricted)
+in the codebase to find where it appears. Change uses of
+`#![feature(XXX)]` from the stdlib and rustc crates to be
+`#![cfg_attr(stage0, feature(XXX))]`. This includes the feature-gate
+only for stage0, which is built using the current beta (this is
+needed because the feature is still unstable in the current beta).
+
+Also, remove those strings from any tests. If there are tests
+specifically targeting the feature-gate (i.e., testing that the
+feature-gate is required to use the feature, but nothing else),
+simply remove the test.
+
+### Do not require the feature-gate to use the feature
+
+Most importantly, remove the code which flags an error if the
+feature-gate is not present (since the feature is now considered
+stable). If the feature can be detected because it employs some
+new syntax, then a common place for that code to be is in the
+same `feature_gate.rs`. For example, you might see code like this:
+
+    ```
+    gate_feature_post!(&self, pub_restricted, span,
+     "`pub(restricted)` syntax is experimental");
+    ```
+
+This `gate_feature_post!` macro prints an error if the
+`pub_restricted` feature is not enabled. It is not needed
+now that `#[pub_restricted]` is stable.
+
+For more subtle features, you may find code like this:
+
+    ```
+    if self.tcx.sess.features.borrow().pub_restricted { /* XXX */ }
+    ```
+
+This `pub_restricted` field (obviously named after the feature)
+would ordinarily be false if the feature flag is not present
+and true if it is. So transform the code to assume that the field
+is true. In this case, that would mean removing the `if` and
+leaving just the `/* XXX */`.
+
+    ```
+    if self.tcx.sess.features.borrow().pub_restricted { /* XXX */ }
+    = becomes ==>
+    /* XXX */
+
+    if self.tcx.sess.features.borrow().pub_restricted && something { /* XXX */ }
+    = becomes ==>
+    if something { /* XXX */ }
+    ```
+
+## Updating documentation
+
+If any documentation for this feature exists, it should be
+in the `Unstable Book`, located at `src/doc/unstable-book`.
```

Comment:
Could you make these into links to https://doc.rust-lang.org/unstable-book/index.html and https://github.com/rust-lang/rust/tree/master/src/doc/unstable-book, respectively?